app.py

```diff
@@ -0,0 +1,423 @@
+        """
+        Called when an item is about to be exported and a path needs to be computed.
+ 
+        :param session_id: String which identifies which export session is being referred to.
+                           This parameter makes it possible to distinguish between different 
+                           export sessions running if this is needed (typically only needed for
+                           expert use cases).
+
+        :param info: Dictionary with a number of parameters:
+        
+           destinationHost: Host name where the exported files will be written to.
+           destinationPath: Export path root.
+           namePattern:     List of optional naming tokens.
+           resolvedPath:    Full file pattern that will be exported with all the tokens resolved.
+           name:            Name of the exported asset.
+           sequenceName:    Name of the sequence the asset is part of.
+           shotName:        Name of the shot the asset is part of.
+           assetType:       Type of exported asset. ( 'video', 'audio', 'batch', 'openClip', 'batchOpenClip' )
+           width:           Frame width of the exported asset.
+           height:          Frame height of the exported asset.
+           aspectRatio:     Frame aspect ratio of the exported asset.
+           depth:           Frame depth of the exported asset. ( '8-bits', '10-bits', '12-bits', '16 fp' )
+           scanFormat:      Scan format of the exported asset. ( 'FILED_1', 'FIELD_2', 'PROGRESSIVE' )
+           fps:             Frame rate of exported asset.
+           sequenceFps:     Frame rate of the sequence the asset is part of.
+           sourceIn:        Source in point in frame and asset frame rate.
+           sourceOut:       Source out point in frame and asset frame rate.
+           recordIn:        Record in point in frame and sequence frame rate.
+           recordOut:       Record out point in frame and sequence frame rate.
+           track:           ID of the sequence's track that contains the asset.
+           trackName:       Name of the sequence's track that contains the asset.
+           segmentIndex:    Asset index (1 based) in the track.
+           versionName:     Current version name of export (Empty if unversioned).
+           versionNumber:   Current version number of export (0 if unversioned).        
+        """
+        if info.get("assetType") != "video":
+            # the review system ignores any other assets. The export profiles are defined
+            # in the app's settings hook, so technically there shouldn't be any other items
+            # generated - but just in case there are (because of customizations), we'll simply
+            # ignore these.
+            return
+        
+        # ensure each quicktime gets a unique name
+        info["resolvedPath"] = "shotgun_%s.mov" % uuid.uuid4().hex
+         
+        
+    def register_post_asset_job(self, session_id, info):
```

Comment:
Wonder if this method should be renamed as register_shotgun_update_job ? The current name is very generic, suggesting it could register any job, whereas what it does is very specific ?